x/evm/types/msg.go

```diff
@@ -412,26 +375,24 @@ func (msg *MsgEthereumTx) BuildTx(b client.TxBuilder, evmDenom string) (authsign
 		return nil, err
 	}
 
-	txData, err := UnpackTxData(msg.Data)
-	if err != nil {
-		return nil, err
-	}
 	fees := make(sdk.Coins, 0)
-	feeAmt := sdkmath.NewIntFromBigInt(txData.Fee())
+	fee := msg.GetFee()
+	feeAmt := sdkmath.NewIntFromBigInt(fee)
 	if feeAmt.Sign() > 0 {
 		fees = append(fees, sdk.NewCoin(evmDenom, feeAmt))
 	}
 
 	builder.SetExtensionOptions(option)
 
-	err = builder.SetMsgs(msg)
-	if err != nil {
+	if err := builder.SetMsgs(&MsgEthereumTx{
+		From: msg.From,
+		Raw:  msg.Raw,
```

Comment:
not sure how sdk handle anyTxData after we remove from FromEthereumTx

Reply:
WDYM?

Reply:
just think cosmos batch tx wont work without pass data like before

Reply:
but the integration test do works?

Reply:
yea raw works, seems no need [PackTxData](https://github.com/crypto-org-chain/ethermint/blob/6c734fbb90d1bc8e31ae69ed64407ed3853ad9a3/x/evm/types/codec.go#L79) now

Reply:
yeah, let's keep it for the unit test for now.